File: Mavim-services/Topic/Mavim.Manager.Api.Topic.Repository/v1/Mappers/TextFieldMapper.cs
using Mavim.Manager.Api.Topic.Repository.Interfaces.v1.enums;
using Mavim.Manager.Api.Topic.Repository.Interfaces.v1.Fields;
using Mavim.Manager.Api.Topic.Repository.v1.Fields;
using Mavim.Manager.Api.Topic.Repository.v1.Mappers.Abstract;
using System;

namespace Mavim.Manager.Api.Topic.Repository.v1.Mappers
{
    internal class TextFieldMapper : GenericTextFieldMapper<Model.ISingleValueSimpleField, ISingleTextField>
    {
        /// <summary>
        /// Gets the generic mapped text service field.
        /// </summary>
        /// <param name="field">The field.</param>
        /// <returns></returns>
        protected override ISingleTextField GetGenericMappedServiceField(Model.ISingleValueSimpleField field)
        {
            if (field == null) throw new ArgumentNullException(nameof(field));

            return new SingleTextField
            {
                FieldId = field.FieldDefinition.ID,
                FieldName = field.FieldDefinition.Name,
                SetOrder = field.FieldSetDefinition.OrderNumber,
                Order = field.OrderNumber,
                FieldSetId = field.FieldSetDefinition.ID,
                SetName = field.FieldSetDefinition?.Name ?? string.Empty,
                FieldValue = field.FieldValue?.ToString(),
                FieldValueType = FieldType.Text,
                Required = field.FieldDefinition.Required,
                Readonly = field.ReadOnly
            };
        }

        /// <summary>
        /// Gets the generic mapped text repo field.
        /// </summary>
        /// <param name="field">The field.</param>
        /// <param name="simpleField">The simple field.</param>
        /// <returns></returns>
        /// <exception cref="ArgumentNullException">field</exception>
        protected override object[] GetGenericMappedRepoField(ISingleTextField field, Model.ISimpleField simpleField = null)
        {
            return new object[] { field?.FieldValue };
        }
    }
}

File: Mavim-apps/Mavim.frontend/projects/mavim-app/src/shared/field/mappers/textfield.mapper.ts
import { Field } from '../models/field.model';
import { SingleTextField } from '../models/api';
import { GenericTextFieldMapper } from './abstract/generic-text-field.mapper';
import { mapToApiFieldType, mapToFieldType } from './fieldvaluetype';

export class TextFieldMapper extends GenericTextFieldMapper<SingleTextField> {
	public get getUrlTypeName(): string {
		return 'text';
	}

	public getMappedField(field: SingleTextField): Field {
		return this.mapField(field);
	}

	public getMappedApiField(field: Field): SingleTextField {
		return this.mapApiField(field);
	}

	protected mapApiField(field: Field): SingleTextField {
		return {
			...field,
			fieldValueType: mapToApiFieldType(field.fieldValueType),
			data: field.data ? this.mapToText(field.data[0]) : undefined
		};
	}

	protected mapField(field: SingleTextField): Field {
		return {
			...field,
			fieldValueType: mapToFieldType(field.fieldValueType),
			options: undefined,
			data: [this.mapToString(field.data)]
		} as Field;
	}
}
